common/kotlinx-coroutines-core-common/src/JobSupport.kt

```diff
@@ -1029,7 +1034,7 @@ public open class JobSupport constructor(active: Boolean) : Job, ChildJob, Paren
         private val job: JobSupport
     ) : CancellableContinuationImpl<T>(delegate, MODE_CANCELLABLE) {
         override fun getContinuationCancellationCause(parent: Job): Throwable {
-            val state = job.state
+            val state = job.state.unboxState()
```

Comment:
Unboxing is not needed here and even harmful, since this code would not work problem in case when `Finishing` instance was boxed inside.

Reply:
You are right, this one is unnecessary.
`Finishing` state can be boxed. I wanted to add `assert` (one with -ea), but it's JVM only